pyomo/contrib/solver/solvers/highs.py

```diff
@@ -96,6 +96,121 @@ def update(self):
         self.highs.changeColCost(col_ndx, value(self.expr))
 
 
+class _MutableQuadraticCoefficient:
+    def __init__(self, expr, row_idx, col_idx):
+        self.expr = expr
+        self.row_idx = row_idx
+        self.col_idx = col_idx
+
+
+class _MutableObjective:
+    def __init__(self, highs, constant, linear_coefs, quadratic_coefs):
+        self.highs = highs
+        self.constant = constant
+        self.linear_coefs = linear_coefs
+        self.quadratic_coefs = quadratic_coefs
+        self.last_quadratic_coef_values = [value(i.expr) for i in self.quadratic_coefs]
+        # Store the quadratic coefficients in dictionary format
+        self.quad_coef_dict = {}
+        self._initialize_quad_coef_dict()
+        # Flag to force first update of quadratic coefficients
+        self._first_update = True
+
+    def _initialize_quad_coef_dict(self):
+        for coef in self.quadratic_coefs:
+            v1_ndx = coef.row_idx
+            v2_ndx = coef.col_idx
+            # Ensure we're storing the lower triangular part
+            row = max(v1_ndx, v2_ndx)
+            col = min(v1_ndx, v2_ndx)
+
+            coef_val = value(coef.expr)
+            # Adjust for diagonal elements
+            if v1_ndx == v2_ndx:
+                coef_val *= 2.0
+
+            self.quad_coef_dict[(row, col)] = coef_val
+
+    def update(self):
+        """
+        Update the quadratic objective expression.
+        """
+        needs_quadratic_update = self._first_update
+
+        self.constant.update()
+        for coef in self.linear_coefs:
+            coef.update()
+
+        for ndx, coef in enumerate(self.quadratic_coefs):
+            current_val = value(coef.expr)
+            if current_val != self.last_quadratic_coef_values[ndx]:
```

Comment:
If the set of quadratic variables changes (e.g., the objective changes from `x**2 + y**2` to `x**2 + z**2`) will `ndx` be in `self.last_quadratic_coef_values`?

Reply:
Oh, this should be fine because `_set_objective` should have been called when the objective changed. Can you just make sure there is a test that validates this? There might already be one. I don't remember.

Reply:
I have changed that logic. The test that deletes x1 should cover this now.